rascal-textmate-core/src/main/rascal/lang/textmate/NameGeneration.rsc

```diff
@@ -0,0 +1,53 @@
+@synoposis{
+    Types and functions to generate names for TextMate rules
+}
+
+module lang::textmate::NameGeneration
+
+import Grammar;
+import ParseTree;
+import String;
+import lang::rascal::format::Grammar;
+
+data NameGeneration              // Given a production `p` of the form `prod(label(l, sort(s)), _, _)`... 
```

Comment:
do we need to have this parameter? can we not just commit to the short names, and if needed, get back the long names from history?

my goal is to reduce the size of the project.

Reply:
The long names help me with testing and debugging, so I'd like to keep the option for now